### PR TITLE
CDAP-1164 fix the way we decode query results from hive for

### PR DESCRIPTION
**Pull request by albertshau**
byte arrays.

### CODE REVIEW

**Comment by gokulavasan** [comment]
LGTM :+1:

**Comment by albertshau** [comment]
https://builds.cask.co/browse/CDAP-RBT63

File: cdap-explore-client/src/main/java/co/cask/cdap/explore/client/AbstractExploreClient.java [hunk 1]
```diff
@@ -396,6 +396,22 @@ protected QueryResult computeNext() {
...
             columnValue = newColumnValue;
+          } else if ("array<tinyint>".equals(columnType)) {
+            // in some versions of hive, a byte[] gets translated to array<tinyint> instead of binary.
+            // weirdly enough, in our unit tests, if java6 is used, byte[] fields get changed to array<tinyint>
+            // but if java7 is used, byte[] fields get changed to binary...
+            // and on top of that it decides to return the byte array as a string... like "[98,111,98]".
+            if (columnValue instanceof String) {
+              String byteArrAsString = (String) columnValue;
+              // strip brackets. [98,111,98] -> 98,111,98
+              byteArrAsString = byteArrAsString.substring(1, byteArrAsString.length() - 1);
```
Comment (chtyim) [inline cdap-explore-client/src/main/java/co/cask/cdap/explore/client/AbstractExploreClient.java]: This is pretty hacky. Actually you can just GSON decode the string to `byte[]`.

**Comment by chtyim** [comment]
LGTM

**Pull request merged**